tests/Ownership/Readmodels/OwnershipSearchProjectorTest.php

```diff
@@ -42,26 +45,83 @@ public function it_handles_ownership_requested(): void
             'google-oauth2|102486314601596809843'
         );
 
-        $recordedOn = RecordedOn::fromBroadwayDateTime(DateTime::fromString('2024-02-19T14:15:16Z'));
-        $domainMessage = new DomainMessage(
-            $ownershipId,
-            0,
-            new Metadata(),
-            $ownershipRequested,
-            $recordedOn->toBroadwayDateTime()
-        );
-
         $ownershipItem = new OwnershipItem(
             $ownershipRequested->getId(),
             $ownershipRequested->getItemId(),
             $ownershipRequested->getItemType(),
             $ownershipRequested->getOwnerId(),
+            OwnershipState::requested()->toString()
         );
 
         $this->ownershipSearchRepository->expects($this->once())
             ->method('save')
             ->with($ownershipItem);
 
-        $this->ownershipSearchProjector->handle($domainMessage);
+        $this->ownershipSearchProjector->handle($this->createDomainMessage($ownershipRequested));
+    }
+
+    /**
+     * @test
+     */
+    public function it_handles_ownership_approved(): void
+    {
+        $ownershipApproved = new OwnershipApproved('e6e1f3a0-3e5e-4b3e-8e3e-3f3e3e3e3e3e');
+
+        $this->ownershipSearchRepository->expects($this->once())
+            ->method('updateState')
+            ->with(
+                $ownershipApproved->getId(),
+                OwnershipState::approved()
+            );
+
+        $this->ownershipSearchProjector->handle($this->createDomainMessage($ownershipApproved));
+    }
+
+    /**
+     * @test
+     */
+    public function it_handles_ownership_rejected(): void
+    {
+        $ownershipRejected = new OwnershipRejected('e6e1f3a0-3e5e-4b3e-8e3e-3f3e3e3e3e3e');
+
+        $this->ownershipSearchRepository->expects($this->once())
+            ->method('updateState')
+            ->with(
+                $ownershipRejected->getId(),
+                OwnershipState::rejected()
+            );
+
+        $this->ownershipSearchProjector->handle($this->createDomainMessage($ownershipRejected));
+    }
+
+    /**
+     * @test
+     */
+    public function it_handles_ownership_deleted(): void
+    {
+        $ownershipDeleted = new OwnershipDeleted('e6e1f3a0-3e5e-4b3e-8e3e-3f3e3e3e3e3e');
+
+        $this->ownershipSearchRepository->expects($this->once())
+            ->method('updateState')
+            ->with(
+                $ownershipDeleted->getId(),
+                OwnershipState::deleted()
+            );
+
+        $this->ownershipSearchProjector->handle($this->createDomainMessage($ownershipDeleted));
+    }
+
+    /**
+     * @param OwnershipRequested|OwnershipApproved|OwnershipRejected|OwnershipDeleted $event
```

Comment:
This is the only part of the PR, that made me doubt, Chaining 4 classes seems a bit too much to me.
But since this is the only place it's needed, introducing a BaseClass would not be a good idea.

Reply:
I could just remove the type hint because it is inside a unit test. In a normal class I would make a factory method for all the different types, but that is overkill here.